### PR TITLE
libmpdec 2.4.2 (new formula)

### PR DESCRIPTION
**Pull request by lpinca**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Previous attempts in chronological order:

- https://github.com/Homebrew/legacy-homebrew/pull/46545
- https://github.com/Homebrew/legacy-homebrew/pull/48557
- https://github.com/Homebrew/legacy-homebrew/pull/49756

### CODE REVIEW

File: Formula/libmpdec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+class Libmpdec < Formula
+  desc "Library for decimal floating point arithmetic"
+  homepage "https://www.bytereef.org/mpdecimal/"
+  url "https://www.bytereef.org/software/mpdecimal/releases/mpdecimal-2.4.2.tar.gz"
+  sha256 "83c628b90f009470981cf084c5418329c88b19835d8af3691b930afccb7d79c7"
+
+  patch do
```
Comment (lpinca) [inline Formula/libmpdec.rb]: This makes `brew audit --new-formula libmpdec` return an error.
I've tried to contact the library author to have the patch applied upstream but no reply yet.

**Comment by fxcoudert** [comment]
Hi @lpinca, thanks for the pull request, we're going to decline for now, and hope the software's author can fix it in a new release. Meanwhile, we can suggest you maintain this in a personal tap: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Comment by lpinca** [comment]
@fxcoudert ok, there quite a few [formulas with a floating path](https://github.com/Homebrew/formula-patches). Can the same be done for this or is the only way to have the patch accepted upstream?

I'm more than happy to submit a PR to https://github.com/Homebrew/formula-patches with the required patch.

**Comment by lpinca** [comment]
@skrah, sorry to mention you here, but I would really like to have this on homebrew.

**Comment by skrah** [comment]
@lpinca  I don't have OSX.  I've already made one `libmpdec` release with a build "fix" for OSX because someone said that `.so` is supported on OSX.

I'll make a new release with `.dylib` support if and only if:

At least one homebrew maintainer confirms that (`xnd` is my new project) this

https://github.com/plures/xnd/blob/c85da07a36a59d07eedda6fb3607a7afa4354b39/configure.ac#L14

is the correct way of configuring for OSX. It works perfectly on Travis-CI buildbots and on conda.

@fxcoudert If you have time, would you mind taking a short look?

**Comment by skrah** [comment]
If I get confirmation, I'll use the exact same way for `libmpdec`.

**Comment by fxcoudert** [comment]
@skrah this seems correct, yes

**Pull request closed without merging**